include/zephyr/fs/nvs.h

```diff
@@ -164,6 +164,26 @@ ssize_t nvs_read_hist(struct nvs_fs *fs, uint16_t id, void *data, size_t len, ui
  */
 ssize_t nvs_calc_free_space(struct nvs_fs *fs);
 
+/**
+ * @brief Tell how many contiguous free space remains in the currently active NVS sector.
+ *
+ * @param fs Pointer to the file system.
+ *
+ * @return Number of free bytes.
+ */
+size_t nvs_sector_max_data_size(struct nvs_fs *fs);
+
+/**
+ * @brief Close the currently active sector and switch to the next one.
+ *
+ * @note The garbage collector is called on the new sector.
```

Comment:
This should have a warning, something like "this routine breaks the aim of nvs to avoid any unnecessary flash erases, using this routine extensively can result in premature failure of the flash device".

Reply:
Warning added.